### PR TITLE
Fix the uniqueEntries filtering

### PR DESCRIPTION
**Pull request by stekaz**
The UniqueEntriesEvaluator does not seem to correctly detect non-unique combinations, at least in my testing. This fix ensures that the combination of values for the given keys is unique across all objects in the array.

### CODE REVIEW

**Comment by nvnieuwk** [comment]
Hi sorry for taking so long to address this PR. I'll check it out now. Can you also update the changelog with this change?

**Review by nvnieuwk — APPROVED** [review]
This change seems reasonable! Thanks! I'll merge this in once the changelog has been updated :)

**Comment by stekaz** [comment]
No worries at all! I have updated the changelog to reflect the changes. Please let me know if there's anything else that needs adjustment, thanks!

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,5 +1,11 @@
 # nextflow-io/nf-schema: Changelog
 
+# Unreleased
+
+## Bug fixes
+
+1. Resolved an issue where the UniqueEntriesEvaluator did not correctly detect non-unique combinations.
```
Comment (nvnieuwk) [inline CHANGELOG.md:7]: Could you add this to the `v2.1.1` changelog? That's the next version :p

Comment (stekaz) [reply CHANGELOG.md]: Whoops, sorry..! Yes you're right. Should be good now hopefully.

**Comment by nvnieuwk** [comment]
Thank you! :partying_face:

**Pull request merged**